graphql_compiler/compiler/ir_sanity_checks.py

```diff
@@ -139,34 +176,33 @@ def _sanity_check_mark_location_preceding_optional_traverse(ir_blocks):
 
 def _sanity_check_every_location_is_marked(ir_blocks):
     """Ensure that every new location is marked with a MarkLocation block."""
-    # Exactly one MarkLocation block is found between a QueryRoot / Traverse / Recurse block,
-    # and the first subsequent Traverse, Recurse, Backtrack or ConstructResult block.
+    # Exactly one MarkLocation block is found between any block that starts an interval of blocks
+    # that all affect the same query position, and the first subsequent block that affects a
+    # different position in the query. Such intervals include the following examples:
+    # - from Fold to Unfold
+    # - from QueryRoot to Traverse/Recurse
+    # - from one Traverse to the next Traverse
+    # - from Traverse to Backtrack
     found_start_block = False
-    found_fold_block = False
     mark_location_blocks = 0
```

Comment:
I would call this something like `num_marklocation_blocks`.

Reply:
Renamed to `mark_location_blocks_count`, is that acceptable?